src/microbe_stage/systems/EngulfedDigestionSystem.cs

```diff
@@ -240,6 +247,9 @@ protected override void Update(float delta, in Entity entity)
                 if (totalAmountLeft <= 0 || engulfable.DigestedAmount >= Constants.FULLY_DIGESTED_LIMIT)
                 {
                     engulfable.PhagocytosisStep = PhagocytosisPhase.Digested;
+
+                    if (entity.Has<PlayerMarker>() && engulfedObject.Has<MicrobeControl>())
```

Comment:
Is it necessary to check that the object has microbe control here? Only entities that have a bunch of the microbe components can be processed by this system, so it is 100% known that entity is a valid microbe when code execution reaches here. Also I'd put the player check outside the loop all the way up on line 79 to just check the component existing once at the start of this method instead for each engulfed object.

Reply:
From my testing, engulfing those floating organelle leftover bits also triggered this, which is why I added this check here

Reply:
Ah yes, of course I was partly mixing up the current entity and the engulf target in my head. In case of the target, you should check for CellProperties, that's my goto property to check if I need to know if something is a microbe or not.

Reply:
Got it